interview-questions.md

```diff
@@ -8,36 +8,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: To fix this mistake, generate a migrations file named specifically what your intentions are to do to. After generating the migrations file, need to run $rails db: migrate which creates a new file. Within that file and within that method, use add_column :table_name, :column_name, :datatype. :table_name specifies which Model you are targeting, :column_name correlates with what you want to add, since it is missing a foreign key, it will be :cohort_id. This will be on the student model.
 
-Researched answer:
+Researched answer: To fix this mistake, open the terminal make sure you are in the correct file and run command $rails g migration, name the file name that starts with the action that describes the purpose of the migration i.e add_column_to_list. Afterwards, a file will be added to the migrate directory and within that file there is a def change method already built. Next, within that method, we want to add a column, add_column :table_name, :column_name, :datatype. :table_name specifies which table we are adding too (:students), :column_name is the name of the new column (:cohort_id), and :datatype is a primitive data type (integer). Since the cohort model has_many students the foreign key will be :cohort_id because it is specifying students belongs_to cohort. The foreign key will also be on the Student model. After checking the migrations file, if it looks correct run db:migrate then it will show up on the schema file.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The RESTful routes that must take a parameter is show, edit, update, destroy. For all of these RESTful routes the param is needed in order to specify which primary key or ID is needed to specify in the url. 
 
-Researched answer:
+Researched answer: The RESTful routes that must take a parameter are show, edit, create, update, and destroy. Within the Show method, the param is used to show one item in the database by accessing the ID. Edit, will use a form for the specified id you want to edit. Create will call upon the strong params method. Update will update the specified item in the database by accessing the ID. Destroy will delete the specified item in the database by accessing the ID.
```

Comment:
Good thought with create using the strong params, but I'd say params and strong params are separate things in the scope of this question